python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -626,6 +627,24 @@ def __set__(self, obj, _val):
         GangaObject.__setattr__(raw_obj, attr_name, final_val)
 
 
+def proxy_wrap(f):
+    # type: (Callable) -> Callable
+    """
+    A decorator to strip the proxy from all incoming arguments
+    (including ``self`` if it's a method) and add one to the return
+    value.
+    """
+
+    @functools.wraps(f)
+    def proxy_wrapped(*args, **kwargs):
+        s_args = [stripProxy(a) for a in args]
```

Comment:
indeed, you beat me to it @milliams. This is the beauty (IMO) of the system that I am using. The proxy is just an outermost layer that object cross. e.g imagine a list of objects, as you fill that list the objects are unwrapped and stored in the list without proxies. Their proxies are applied again when you get them out.